To: vim_dev@googlegroups.com
Subject: Patch 7.3.425
Fcc: outbox
From: Bram Moolenaar <Bram@moolenaar.net>
Mime-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit
------------

Patch 7.3.425 (after 7.3.265)
Problem:    Search history lines are duplicated. (Edwin Steiner)
Solution:   Convert separator character from space to NUL.
Files:      src/ex_getln.c


*** ../vim-7.3.424/src/ex_getln.c	2012-01-26 13:01:54.000000000 +0100
--- src/ex_getln.c	2012-02-04 22:27:43.000000000 +0100
***************
*** 6027,6034 ****
  	val = viminfo_readstring(virp, 1, TRUE);
  	if (val != NULL && *val != NUL)
  	{
  	    if (!in_history(type, val + (type == HIST_SEARCH),
! 						  viminfo_add_at_front, *val))
  	    {
  		/* Need to re-allocate to append the separator byte. */
  		len = STRLEN(val);
--- 6027,6036 ----
  	val = viminfo_readstring(virp, 1, TRUE);
  	if (val != NULL && *val != NUL)
  	{
+ 	    int sep = (*val == ' ' ? NUL : *val);
+ 
  	    if (!in_history(type, val + (type == HIST_SEARCH),
! 						   viminfo_add_at_front, sep))
  	    {
  		/* Need to re-allocate to append the separator byte. */
  		len = STRLEN(val);
***************
*** 6040,6046 ****
  			/* Search entry: Move the separator from the first
  			 * column to after the NUL. */
  			mch_memmove(p, val + 1, (size_t)len);
! 			p[len] = (*val == ' ' ? NUL : *val);
  		    }
  		    else
  		    {
--- 6042,6048 ----
  			/* Search entry: Move the separator from the first
  			 * column to after the NUL. */
  			mch_memmove(p, val + 1, (size_t)len);
! 			p[len] = sep;
  		    }
  		    else
  		    {
*** ../vim-7.3.424/src/version.c	2012-02-04 22:01:44.000000000 +0100
--- src/version.c	2012-02-04 22:30:36.000000000 +0100
***************
*** 716,717 ****
--- 716,719 ----
  {   /* Add new patch number below this line */
+ /**/
+     425,
  /**/

-- 
"So this is it," said Arthur, "we are going to die."
"Yes," said Ford, "except...no!  Wait a minute!"  He suddenly lunged across
the chamber at something behind Arthur's line of vision.  "What's this
switch?" he cried.
"What?   Where?" cried Arthur, twisting around.
"No, I was only fooling," said Ford, "we are going to die after all."
		-- Douglas Adams, "The Hitchhiker's Guide to the Galaxy"

 /// Bram Moolenaar -- Bram@Moolenaar.net -- http://www.Moolenaar.net   \\\
///        sponsor Vim, vote for features -- http://www.Vim.org/sponsor/ \\\
\\\  an exciting new programming language -- http://www.Zimbu.org        ///
 \\\            help me help AIDS victims -- http://ICCF-Holland.org    ///
